climpred/smoothing.py

```diff
@@ -0,0 +1,248 @@
+        ds (xr.object): input. xr.DataArray prefered.
+        coarsen_dict (dict): coarsen spatial latitudes.
+        how (str): aggregation type for coarsening. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): boxsize-regridded input
+
+    """
+    if coarsen_dict is None:
+        # guess spatial dims
+        spatial_dims_to_smooth = list(ds.dims)
+        for dim in ['time', 'lead', 'member', 'init']:
+            if dim in ds.dims:
+                spatial_dims_to_smooth.remove(dim)
+        # write coarsen to dict to coarsen similar to 5x5 degree
+        coarsen_dict = dict()
+        step = 2
+        print(
+            f'no coarsen_dict given. created for dims \
+            {spatial_dims_to_smooth} with step {step}'
+        )
+        if len(spatial_dims_to_smooth) == 2:
+            for dim in spatial_dims_to_smooth:
+                coarsen_dict[dim] = step
+        else:
+            raise ValueError(
+                f'Tried to guess coarsen_dict. Found {spatial_dims_to_smooth},\
+                  but length != 2. Please provide coarsen_dict.'
+            )
+
+    # check whether coarsen dims are possible
+    for dim in coarsen_dict:
+        if dim not in ds.dims:
+            raise ValueError(dim, 'not in ds')
+        else:
+            if ds[dim].size % coarsen_dict[dim] != 0:
+                raise ValueError(
+                    coarsen_dict[dim], 'does not divide evenly', ds[dim].size, 'in', dim
+                )
+    ds_out = getattr(ds.coarsen(coarsen_dict), how)()
+    return ds_out
+
+
+def temporal_smoothing(ds, smooth_dict=None, how='mean', rename_dim=True):
+    """Apply temporal smoothing by creating rolling smooth-timestep means.
+
+    Reference:
+      * Goddard, L., A. Kumar, A. Solomon, D. Smith, G. Boer, P.
+            Gonzalez, V. Kharin, et al. “A Verification Framework for
+            Interannual-to-Decadal Predictions Experiments.” Climate
+            Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+            https://doi.org/10/f4jjvf.
+
+    Args:
+        ds (xr.object): input.
+        smooth_dict (dict): length of smoothing of timesteps.
+                      Defaults to {'time':4} (see Goddard et al. 2013).
+        how (str): aggregation type for smoothing. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): input with `smooth` timesteps less and
+                                 labeling '1-(smooth-1)','...', ... .
+
+    """
+    # unpack dict
+    if smooth_dict is None:
+        smooth_dict = {'time': 4}
+    if not ('time' not in smooth_dict or 'lead' not in smooth_dict):
+        raise ValueError('smooth_dict doesnt contain a time dimension.', smooth_dict)
+    smooth = list(smooth_dict.values())[0]
+    dim = list(smooth_dict.keys())[0]
+    # fix to smooth either lead or time depending
+    time_dims = ['time', 'lead']
+    if dim not in ds.dims:
+        time_dims.remove(dim)
+        dim = time_dims[0]
+        smooth_dict = {dim: smooth}
```

Comment:
I think lines 201-205 can be elegantly compressed with a dict comprehension since we don't use time_dims anywhere else
`smooth_dict = {dim: smooth for dim in ['time', 'lead'] if dim not in ds.dims}`

Reply:
this would change the logic